---
title: "Class 13: Genome Informatics 1"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Genome Informatics- Using Information from an Online Database

We gathered this information from the *add website* Ensembl page. The idea is to figure out a percentage of our population that has 
After we put the file into our directory, we need to read it in. 

```{r}
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```
Now we look at the numbers of each genotype. 
```{r}
table(mxl$Genotype..forward.strand.)
```

Calculating the percentage of G|G in the total popuation.
```{r}
table(mxl$Genotype..forward.strand.) / nrow(mxl) * 100
```

##Initial RNA Seq Analysis

We'll download the fasta files from the class website. We'll be using unix commands to look at this data because the files are so large. 
  head ~Downloads/HG00109_1.fastq
  head allows us to look at our data. 
  
This gives us a unique identifier, our code, and then the ASCII base qualities. 

##Quality Scores
```{r}
#install seqinr and gtools
library(seqinr)
library(gtools)
chars <- s2c("DDDDCDEDCDDDDBBDDDCC@")
chars
phred <- asc( chars ) - 33
phred
```

Now we'll be trying out Galaxy. (Using a previously set up IP address).

##Population Scale Analysis

```{r}
expr <- read.table("rs8067378_ENSG00000172057.6.txt")
head(expr)
```
```{r}
table(expr$geno)
```

```{r}
inds.gg <- expr$geno == "G/G"
summary(expr[inds.gg,"exp"])
```

```{r}
inds.ag <- expr$geno == "A/G"
summary(expr[inds.ag, "exp"])
```

```{r}
inds.aa <- expr$geno == "A/A"
summary(expr[inds.aa, "exp"])
```

We have all of this information, now we'd like a much nicer way to display it. 
Let's use a boxplot!

```{r}
boxplot(exp ~ geno, data=expr)
```

Now it's time to change the graphics. 
```{r}
boxplot(exp ~ geno, data=expr, notch = TRUE)
```

There are other graphs we can use. (Maybe try ggplot).

##Experimenting with "GGPlot"" Package


